raiden/raiden_service.py

```diff
@@ -325,6 +326,13 @@ def handle_state_change(self, state_change, block_number=None):
         if block_number is None:
             block_number = self.get_block_number()
 
+        # Take a snapshot every SNAPSHOT_BLOCK_COUNT
+        # TODO: Gather more data about storage requirements
+        # and update the value to specify how often we need
+        # capturing a snapshot should take place
+        if block_number % SNAPSHOT_BLOCK_COUNT == 0:
```

Comment:
if alarm misses a block, it may skip the snapshot

Reply:
Humm, true, we need a fix for this, what about: (?)

```python
new_group = block_number // SNAPSHOT_BLOCK_COUNT
if new_group > old_group:
  snapshot()
  old_group = new_group
```

Reply:
I've implemented this idea... as it should resolve the issue. Also restored the "snapshot" group when raiden service is started.